src/kit.ts

```diff
@@ -602,18 +571,38 @@ const VsGenerators: {[key: string]: string} = {
   16: 'Visual Studio 16 2019'
 };
 
-async function varsForVSInstallation(inst: VSInstallation, arch: string): Promise<Map<string, string>|null> {
-  console.log(`varsForVSInstallation path:'${inst.installationPath}' version:${inst.installationVersion} arch:${arch}`);
+async function varsForVSInstallation(inst: VSInstallation, hostArch: string, targetArch: string): Promise<Map<string, string>|null> {
+  console.log(`varsForVSInstallation path:'${inst.installationPath}' version:${inst.installationVersion} host arch:${hostArch} - target arch:${targetArch}`);
+
+  // TODO: if needed, cover also the scenario of VS older than 2017,
```

Comment:
I thought the older VS versions are handled on line 583.  It looks like it's searching the path you mention in this comment.  Can this TODO be removed?

Reply:
Oh I didn't realize that. It's true. I will remove the comment and also do another fix related to x64 and amd64, because the older vcvars scripts don't recognize x64.